pandas/core/indexing.py

```diff
@@ -1854,7 +1858,7 @@ def _setitem_with_indexer_missing(self, indexer, value):
             if index.is_unique:
                 new_indexer = index.get_indexer([new_index[-1]])
                 if (new_indexer != -1).any():
-                    return self._setitem_with_indexer(new_indexer, value)
+                    return self._setitem_with_indexer(new_indexer, value, name)
```

Comment:
double check me on this, but i think it _might_ be the case that we only get here with `name = "loc"`

Reply:
Yep, you are right. Iloc raises when indexer is a dict. That is the only way we can get in there. Removed it